clj: add page (#4980)
tldr-pages/tldr: pages/common/clj.md

# clj

> Clojure tool to start a REPL or invoke a specific function with data.
> All options can be defined in a `deps.edn` file.
> More information: <https://clojure.org/guides/deps_and_cli>.

- Start a REPL:

`clj`

> - Exec function:

Reviewer: Perhaps this will read better?
```suggestion
- Execute a function:
```

> `clj [clj-opt*] [-A:aliases] [init-opt*]`

Reviewer: Wow, this is a pretty complex example to be starting a page with. Are you aware of our `{{token_syntax}}` we've got documented in our [style guide](https://github.com/tldr-pages/tldr/blob/master/contributing-guides/style-guide.md) at all? The page for `git` is a good example of how it's used.

The same goes for the rest of the page.

Author: oh, my bad. I haven't read that page.
I will change the example to just start a repl without options. that first one is the first example of `clj -h` but I agree it's confusing.

```suggestion
`clj`
```

> `clj -X {{namespace/fn_name}}`

Reviewer: Generally, we prefer to avoid shortening names to aid clarity :-)
```suggestion
`clj -X {{namespace/function_name}}`
```

Also, is there a long form of the argument there? Generally we prefer the long forms of arguments in tldr pages. The page goes for other examples.

> - Run -main function of a namespace:

Reviewer: Perhaps this will read better?
```suggestion
- Run the main function of a specified namespace:
```

`clj -M -m {{namespace}} {{args}}`

> - Prepare (resolve deps, download libraries, make and cache classpaths):

Reviewer: Perhaps this will read better?
```suggestion
- Prepare a project by resolving dependencies, downloading libraries, and making / caching classpaths:
```

`clj -P`

> - Start nREPL server with CIDER middleware:

Reviewer: Perhaps this will read better?
```suggestion
- Start an nREPL server with the CIDER middleware:
```

> `clj -Sdeps '{:deps {nrepl {:mvn/version "0.7.0"} cider/cider-nrepl {:mvn/version "0.25.2"}}}' -m nrepl.cmdline --middleware '["cider.nrepl/cider-middleware"]' --interactive`

Reviewer: What's the difference between this example and the earlier REPL one? This example looks rather long and complicated. Is there a way we could shorten it at all?

Author: This one enables external tools to communicate with the REPL through the nREPL API. The way to make it shorter would be removing CIDER middleware which is nice to have but totally optional for most IDEs/tools.
```suggestion
- Start nREPL server to be used with IDEs/tools:

`clj -Sdeps '{:deps {nrepl/nrepl {:mvn/version "0.7.0"}}}' -m nrepl.cmdline --interactive`
```

Reviewer: Ah, I see. Thanks for the explanation! Let's keep it as-is then.

> - Start REPL for ClojureScript and open browser:

Reviewer: Perhaps this will read better?
```suggestion
- Start REPL for ClojureScript and open a web browser:
```

> - Start REPL for ClojureScript and open a web browser:

Reviewer: Perhaps this will read better?
```suggestion
- Start a REPL for ClojureScript and open a web browser:
```
